authors/apps/articles/views.py

```diff
@@ -199,3 +200,48 @@ def update(self, request, slug, id):
         serializer.is_valid(raise_exception=True)
         serializer.save()
         return Response(serializer.data)
+class LikeDislikeView(CreateAPIView, RetrieveUpdateDeleteArticle):
+    """
+    this class defines the endpoint for like and dislike
+    """
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    serializer_class = LikeDislikeSerializer
+    model = None    # Data Model - Articles or Comments
+    vote_type = None  # Vote type Like/Dislike
+
+    def post(self, request, slug):
+        """
+        This view enables a user to like or dislike an article
+        """
+        # check an article is existing
+        article = get_article(slug)
+        
+        try:
+            # if user has ever voted
+            like_dislike = LikeDislike.objects.get(
+                content_type=ContentType.objects.get_for_model(article),
+                object_id=article.id,
+                user=request.user)
+            # check if the user has liked or disliked the article or comment already
```

Comment:
Is it possible for users to like the comments too according to your comment here?

Reply:
Yea , it is. but that is another task on the PT Board @eduhmik